### DISCUSSION TITLE
Package app as Flatpak

### DISCUSSION THREAD

**Discussion by philipp-tty** [f1multiviewer/issue-tracker#562] [Feature requests] [answered]
Currently, MV is available for .deb and .rpm based distros and ships one universal binary.

Using the universal binary is not ideal and a Flatpak would be greatly appreciated for added compatibility, security and ease of use.

Almost all Linux distros support Flatpaks at this point, which would make the .deb and .rpm packages redundant.

**Comment by rijnhard** [comment]
Please do this, the manual updating on a Home Theatre PC is getting painful...
And flatpak build files are really not too bad, even more complex ones like the [MPV](https://github.com/flathub/io.mpv.Mpv/blob/master/io.mpv.Mpv.yml) (which will be a really good reference) are actually not too bad to work with compared to packaging debs or rpms.

Another good reference would be the [stremio flatpak](https://github.com/Stremio/stremio-service/blob/master/com.stremio.Service.json) (the do things a bit differently though)

**Reply by slyd0x** [reply]
I really would like a flatpak as well. Updating would be so much easier.

**Reply by hpka** [reply]
Perhaps to give this more context:

- Many users have been at least curious about Linux for desktop use in recent times.
- You already have .debs and .rpms for Debian/Ubuntu/etc and Fedora/RHL/etc
- In recent years atomic/immutable desktops are starting to raise in prominence, for desktop purposes often motivated ease of setup and updates. For instance, I'm writing this watching FP1 in Chromium on a [Aurora](https://getaurora.dev/en), by [Universal Blue](https://universal-blue.org/), based on [Fedora](https://fedoraproject.org). In short the _easiest_ way to install onto those is [Flatpak](https://flatpak.org).
- Discover-ability increased through being listed on [Flathub](https://flathub.org/).

Of course what I am saying also applies to [Snaps](https://snapcraft.io) to some extent. You can and should consider doing both. AppImages have also [been requested/discussed](https://github.com/f1multiviewer/issue-tracker/discussions/203).

**Reply by 0x4161726f6e** [reply]
I will add my 2 cents to this.  One of the major benefits of running Linux (Both M$ and Apple are catching on) is a single package manager that updates all software on a computer.  Right now MultiViewer is the only app that on my computer that doesn't update with the rest of my system and it is frustrating to open MutliViewer only to learn I need to do a Windows/MacOS style update dance.  From where I am sitting it would make sense to focus on supporting WinGet, whatever Apple is making, FlatPak, Snap, and AppImage; as these appear to be the application distribution methods of the future.  And then there is the whole argument of making things easier for Normal users (statistical normal) while not training/teaching them to do things in a way that malicious people can take advantage of.

**Comment by LapsTimeOFF** [comment]
Hey, we have already tried packaging MultiViewer to a flatpak package multiple times in the past to no success, we may try again in the near future.

There is also currently a community-maintained AUR package of MultiViewer available

**Reply by birdhimself** [reply]
I hacked together a Flatpak build for MultiViewer a couple of months ago and got it working with Chrome's sandbox disabled (I believe running Chrome's sandbox inside a Flatpak requires some special handling, but I didn't look into it at the time). What was it that wasn't working?

**Reply by JustJoostNL** [reply]
The build just never finishes. 😦

**Comment by LapsTimeOFF** [comment, marked as answer]
Hey, Flatpak builds are now available starting from `v2.8.0`, you can find them on [our downloads page](https://muvi.gg/dl), we may add the app to the Flathub later in time, but this is not a priority as of now.
